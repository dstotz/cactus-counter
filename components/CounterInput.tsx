import { counterTextFormat } from "@/lib/counterTextFormat";
import { useState } from "react";
import { StyleSheet, TextInput, View } from "react-native";

type CounterInputProps = {
  value: number;
  onChange: (newCounters: number) => void;
};

export const CounterInput = ({ value, onChange }: CounterInputProps) => {
  const [lastInputChange, setLastInputChange] = useState<string | undefined>();
  const inputDirty = lastInputChange === value.toString();

  return (
    <View style={styles.container}>
      <TextInput
        caretHidden={true}
        style={styles.textInput}
        value={inputDirty ? value.toString() : counterTextFormat(value)}
        onChange={(e) => {
          const newVal = e.nativeEvent.text;
          onChange(Number(newVal));
          setLastInputChange(newVal);
        }}
      ></TextInput>
    </View>
  );
};

const styles = StyleSheet.create({
  container: {},
  textInput: {
    textAlign: "center",
    fontSize: 32,
    color: "black",
    borderWidth: 1,
    borderColor: "black",
    borderRadius: 8,
    padding: 6,
    width: 165,
    height: 75,
  },
});
