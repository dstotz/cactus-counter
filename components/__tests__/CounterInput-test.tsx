import * as React from "react";
import { render } from "@testing-library/react-native";

import { CounterInput } from "../CounterInput";

it(`renders correctly`, () => {
  const tree = render(<CounterInput value={5} onChange={() => {}} />).toJSON();

  expect(tree).toMatchSnapshot();
});
