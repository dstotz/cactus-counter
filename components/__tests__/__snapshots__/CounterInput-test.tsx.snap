// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={{}}
>
  <TextInput
    caretHidden={true}
    onChange={[Function]}
    style={
      {
        "borderColor": "black",
        "borderRadius": 8,
        "borderWidth": 1,
        "color": "black",
        "fontSize": 32,
        "height": 75,
        "padding": 6,
        "textAlign": "center",
        "width": 165,
      }
    }
    value="5"
  />
</View>
`;
